checkov/secrets/scan_git_history.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+import git
+
+from detect_secrets.core import scan
+from git import InvalidGitRepositoryError, GitCommandError, Commit  # type: ignore
+
+if TYPE_CHECKING:
+    from detect_secrets import SecretsCollection
+
+
+def get_commits(root_folder: str) -> list[Commit] | None:
+    try:
+        repo = git.Repo(root_folder)
+    except InvalidGitRepositoryError:
+        logging.error(f"Folder {root_folder} is not a GIT project")
+        return None
+    return list(repo.iter_commits(repo.active_branch, max_count=7))
+
+
+def scan_history(root_folder: str, secrets: SecretsCollection) -> None:
+    commits = get_commits(root_folder)
+    if not commits:
+        return
+    scanned_file_count = 0
+    skipped_file_count = 0
+
+    # we scan the diff between the commit and the next commit - start from the end
+    for previous_commit_idx in range(len(commits) - 1, 0, -1):
```

Comment:
the current list, for `commits = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]` will return `[9, 8, 7, 6, 5, 4, 3, 2, 1]`
```suggestion
    for previous_commit_idx in range(len(commits) - 1, -1, -1):
```

Reply:
i know. in every iteration, i compare the commit to the one after so I don't want to get  previous_commit_idx = 0
- commits = [0,1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  it will return [1,2, 3, 4, 5, 6, 7, 8, 9,10]